### PR TITLE
[MSPAINT] Fix Bezier behaviour

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-17930](https://jira.reactos.org/browse/CORE-17930)

## Proposed changes

- Add `GetDistancePow2` helper function.
- Calculate the norms of two control points.
- Swap two control points if the norm is less than another norm.

## TODO

- [x] Do tests.

## Screenshot
after

### CODE REVIEW

File: base/applications/mspaint/drawing.cpp [hunk 1]
```diff
@@ -91,14 +91,35 @@ Poly(HDC hdc, POINT * lpPoints, int nCount,  COLORREF fg,  COLORREF bg, int thic
...
 Bezier(HDC hdc, POINT p1, POINT p2, POINT p3, POINT p4, COLORREF color, int thickness)
 {
     HPEN oldPen;
     POINT fourPoints[4];
+    double distance12 = GetDistancePow2(p1, p2);
+    double distance13 = GetDistancePow2(p1, p3);
+    double distance42 = GetDistancePow2(p4, p2);
+    double distance43 = GetDistancePow2(p4, p3);
+    double norm1 = distance42 - distance12;
+    double norm2 = distance43 - distance13;
```
Comment (HBelusca) [inline base/applications/mspaint/drawing.cpp:111]: Why using doubles? The GetDistancePow2 evaluated squared differences of LONG, so the result is naturally a LONG (if casted to a double, the decimal part would always be zero), and then you look at differences of distances, they will still be LONG.

File: base/applications/mspaint/drawing.cpp [hunk 2]
```diff
@@ -91,14 +91,35 @@ Poly(HDC hdc, POINT * lpPoints, int nCount,  COLORREF fg,  COLORREF bg, int thic
     SetROP2(hdc, oldRop);
 }
 
+double GetDistancePow2(POINT p1, POINT p2)
```
Comment (HBelusca) [inline base/applications/mspaint/drawing.cpp:94]: Can return `LONG` instead (see comment below). Also, it could be marked `inline`.

Comment (vicmarcal) [reply base/applications/mspaint/drawing.cpp:94]: At first sight seems it is returning LONG * LONG + LONG * LONG, can we be sure that those LONGs are small enough that the result will fit in a LONG too?

Comment (katahiromz) [reply base/applications/mspaint/drawing.cpp]: Fixed.

File: base/applications/mspaint/drawing.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/drawing.cpp]: Fixed.

File: base/applications/mspaint/drawing.cpp (hunk 2 above)
Comment (katahiromz) [reply base/applications/mspaint/drawing.cpp]: 5000 * 5000 == 25000000.
`MAXLONG` == 4294967295.
Good enough!

**Comment by roybaer** [comment]
To be honest, I don't quite get what this does and more importantly why it does what it does.
Does it try to enforce that a control point is close to an endpoint?
If it does, that would actually limit the tool's functionality: You can no linger create loops using the bezier tool.

**Comment by katahiromz** [comment]
MS mspaint in Windows 10:
image
Sorry. My assumption is wrong. I will close this PR.

**Pull request closed without merging**